src/bionics.cpp

```diff
@@ -430,8 +433,38 @@ bool Character::activate_bionic( int b, bool eff_only )
             }
         }
 
-        const size_t win_h = std::min( static_cast<size_t>( TERMY ), bad.size() + good.size() + 2 );
-        const int win_w = 46;
+        const std::map<vitamin_id, vitamin> &vit_all = vitamin::all();
+        for( const auto &v : vit_all ) {
+            if( !v.second.has_flag( "NO_DISPLAY" ) && !v.second.has_flag( "NOT_IN_BLOOD" ) ) {
+                switch( v.second.type() ) {
+                    case vitamin_type::VITAMIN:
+                        if( v.second.get_string_level( vitamin_get( v.first ) ) != "" ) {
+                            vit.push_back( v.second.get_string_level( vitamin_get( v.first ) ) );
+                        }
+                        break;
+                    case vitamin_type::TOXIN:
+                        if( v.second.get_string_level( vitamin_get( v.first ) ) != "" ) {
+                            tox.push_back( v.second.get_string_level( vitamin_get( v.first ) ) );
+                        }
+                        break;
+                    case vitamin_type::DRUG:
+                        if( v.second.get_string_level( vitamin_get( v.first ) ) != "" ) {
+                            drug.push_back( v.second.get_string_level( vitamin_get( v.first ) ) );
+                        }
+                        break;
+                    default:
+                        if( v.second.get_string_level( vitamin_get( v.first ) ) != "" ) {
+                            cnt.push_back( v.second.get_string_level( vitamin_get( v.first ) ) );
+                        }
+                }
+            }
+        }
+
+        const int effect_spacing = ( good.empty() && bad.empty() ) ? 1 : good.size() + bad.size();
+        const size_t win_h = std::min( static_cast<size_t>( TERMY ),
+                                       effect_spacing + vit.size() + tox.size() + drug.size() + cnt.size() + 2 );
+        //const int win_w = 46;
```

Comment:
Please remove any commented out code, it's no longer useful.
```suggestion
```

Reply:
Will do when figure out how exactly I should optimize Analyzer window width.